### PR TITLE
[RFR] Remove resource

### PR DESCRIPTION
**Pull request by ThieryMichel**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
     "lodash.omit": "4.5.0",
     "lodash.pickby": "4.6.0",
     "material-ui": "0.16.7",
+    "moment": "^2.17.1",
```
Comment (djhi) [inline package.json]: `^` is evil

File: src/api/controller/api/export.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ export async function exportMiddleware(ctx, type) {
         const exportStreamFactory = ctx.getExporter(type);
         const characteristics = await ctx.publishedCharacteristic.find({}).toArray();
         const fields = await ctx.field.find({}).toArray();
-        const publishedDatasetStream = await ctx.publishedDataset.find({}).stream();
+        const publishedDatasetStream = ctx.publishedDataset.getFindAllStream();
```
Comment (djhi) [inline src/api/controller/api/export.js:27]: :+1:

File: src/api/models/publishedDataset.js [hunk 3]
```diff
@@ -4,8 +4,18 @@ import omit from 'lodash.omit';
 
 export default (db) => {
     const collection = db.collection('publishedDataset');
+
     collection.insertBatch = documents => chunk(documents, 100).map(data => collection.insertMany(data));
-    collection.findLimitFromSkip = (limit, skip) => collection.find().skip(skip).limit(limit).toArray();
+
+    collection.findLimitFromSkip = (limit, skip) =>
```
Comment (djhi) [inline src/api/models/publishedDataset.js]: would be better to add an optional `filter` parameter here and use this method in `findPage` imo

File: src/app/js/resource/Detail.js [hunk 4]
```diff
@@ -12,20 +15,30 @@ import {
...
+                <CardActions>
+                    <Link to={{ pathname: '/resource/edit', query: { uri: resource.uri } }}>
+                        <FlatButton className="edit-resource" label={polyglot.t('edit')} primary />
+                    </Link>
+                    <Link to={{ pathname: '/resource/hide', query: { uri: resource.uri } }}>
+                        <FlatButton className="hide-resource" label={polyglot.t('hide')} primary />
+                    </Link>
+                </CardActions>
+            :
+                <span />
```
Comment (djhi) [inline src/app/js/resource/Detail.js]: why not returning `null` ?

File: src/app/js/resource/RemovedDetail.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+};
+
+RemovedDetailComponent.propTypes = {
+    reason: PropTypes.string,
+    removedAt: PropTypes.string,
+    p: polyglotPropTypes.isRequired,
+};
+
+const mapStateToProps = state => ({
+    ...getRemovedData(state),
```
Comment (djhi) [inline src/app/js/resource/RemovedDetail.js]: return `getRemovedData(state)` directly

File: src/app/js/resource/index.spec.js [hunk 6]
```diff
@@ -8,9 +8,12 @@ import reducer, {
     SAVE_RESOURCE,
     SAVE_RESOURCE_SUCCESS,
     SAVE_RESOURCE_ERROR,
+    HIDE_RESOURCE,
+    HIDE_RESOURCE_SUCCESS,
+    HIDE_RESOURCE_ERROR,
 } from './index';
 
-describe('resourceReducer', () => {
+describe.only('resourceReducer', () => {
```
Comment (djhi) [inline src/app/js/resource/index.spec.js]: only spotted

File: src/app/js/resource/routes.js [hunk 7]
```diff
@@ -1,19 +1,51 @@
 import React from 'react';
 
 import Resource from './Resource';
+import { isLoggedIn as selectIsLoggedIn } from '../user';
+
+export const onEnterWithAuthenticationCheck = store => (nextState, replaceState) => {
```
Comment (djhi) [inline src/app/js/resource/routes.js:6]: We should extract this function from `admin/routes.js` and avoid duplication

**Review by djhi — CHANGES_REQUESTED** [review]

File: src/app/js/resource/Detail.js (hunk 4 above)
Comment (ThieryMichel) [reply src/app/js/resource/Detail.js]: Last time I tried something like that it displayed null. Has it changed?

File: src/app/js/resource/routes.js (hunk 7 above)
Comment (ThieryMichel) [reply src/app/js/resource/routes.js:6]: The logic is different, here i do not redirect toward login.

File: src/app/js/resource/RemovedDetail.js (hunk 5 above)
Comment (djhi) [reply src/app/js/resource/RemovedDetail.js]: Forgot this one ?

File: src/app/js/resource/routes.js (hunk 7 above)
Comment (djhi) [reply src/app/js/resource/routes.js:6]: nvm, they really are too different.

**Review by djhi — APPROVED** [review]

**Pull request merged**